File: app/src/main/java/com/example/notphilphil/bob/models/LocationEmployee.java
package com.example.notphilphil.bob.models;

public class LocationEmployee extends User {

    private final int employeeID;
    private final Location location;

    private LocationEmployee(String userID, String name, Location _location) {
        super(userID, name);
        employeeID = 0;
        location = _location;
    }

    public LocationEmployee(String userID, String name) {
        this(userID, name, new Location());
    }

    public String toString() {
        return "My employee ID is " + employeeID + " and my location is: " + location ;
    }
}

File: app/src/main/java/com/example/notphilphil/bob/models/LocationEmployee.kt
package com.example.notphilphil.bob.models

class LocationEmployee private constructor(userID: String, name: String, private val location: Location) : User(userID, name) {

    private val employeeID: Int

    init {
        employeeID = 0
    }

    constructor(userID: String, name: String) : this(userID, name, Location()) {}

    override fun toString(): String {
        return "My employee ID is $employeeID and my location is: $location"
    }
}
